salt/journal/init.sls

```diff
@@ -117,25 +109,39 @@ composer-install:
         {% endif %}
         - cwd: /srv/journal/
         - user: {{ pillar.elife.deploy_user.username }}
-        # to correctly write into var/
-        - umask: 002
         - env:
-            - SYMFONY_ENV: {{ pillar.elife.env }}
             - COMPOSER_DISCARD_CHANGES: 'true'
         - require:
             - file: config-file
-            - journal-cache-clean
             - journal-php-extensions
 
+journal-cache-clean:
+    cmd.run:
+        - name: rm -rf var/cache/
+        - cwd: /srv/journal/
+        - require:
+            - var-directory
+
+journal-cache-warmup:
+    cmd.run:
+        - name: bin/console cache:warmup
+        - cwd: /srv/journal/
+        - user: {{ pillar.elife.webserver.username }}
```

Comment:
executing cache-related creations as `www-data` changes the ownership from `elife:www-data` to `www-data:www-data`.
Given the sticky group bit:
```
ls -ld /srv/journal/var/
drwxrwsr-x 5 www-data www-data 4096 May 31 21:15 /srv/journal/var/
```
all subdirectories and files created in `var/` should inherit the group, hence they will also be `*:www-data`.

The permissions of these files should depend on the umask only. For example:
```
ls -ld /srv/journal/var/cache/prod/journalProdProjectContainer.php
-rw-rw-r-- 1 elife www-data 772 May 31 21:16 /srv/journal/var/cache/prod/journalProdProjectContainer.php
```

Without the umask, this should become:
```
-rw-r--r-- 1 www-data www-data ... /srv/journal/var/cache/prod/journalProdProjectContainer.php
```
which works for nginx but is not writable anymore by the `elife` user if commands need to be run. In fact `rm -rf var/cache` is now run as `root`. Will it cause problems like failing a command run through SSH?

Reply:
The umask has come back so any process should correctly create 664 files.